File: CivitasRuby/lib/sorpresa_pagar_cobrar.rb
# encoding:utf-8
require_relative 'sorpresa'
module Civitas
  class SorpresaPagarCobrar < Sorpresa
    def initialize(texto,valor)
      super(texto)
      @valor = valor
    end
    
    def informe(actual, todos)
      Diario.instance.ocurre_evento("Aplicando sorpresa " + self.to_s + 
        " al jugador " + todos[actual].nombre + " de tipo PagarCobrar")
    end
    
    def aplicar_a_jugador(actual, todos)
      if jugador_correcto(actual,todos)
        informe(actual,todos)
        todos[actual].modificar_saldo(@valor)
      end
    end
    
    def to_s
      "Sorpresa: { \n Texto: #{@texto}  \n Tipo: PagarCobrar   \n Valor: #{@valor} }"
    end
    
    
  end
end

File: CivitasJava/src/Civitas/SorpresaPagarCobrar.java
package Civitas;

import java.util.ArrayList;

public class SorpresaPagarCobrar extends Sorpresa{
    private int valor;
    
    SorpresaPagarCobrar(int valor, String texto){
        super(texto);
        this.valor=valor;
    }
    
    @Override
    void aplicarAJugador(int actual, ArrayList<Jugador> todos){
        if(super.jugadorCorrecto(actual,todos)){
            informe(actual,todos);
            todos.get(actual).modificarSaldo(this.valor);        
        }    
    }
    
    @Override
    public String toString(){
        return "Sorpresa: { \n Texto = " + texto + "\n Tipo = PagarCobrar" + 
                "\n Valor = " + valor + "}";
    }
}
